internal/dag/policy_test.go

```diff
@@ -1217,7 +1217,68 @@ func TestValidateHeaderAlteration(t *testing.T) {
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			got, gotErr := headersPolicyService(test.dhp, test.in, test.dyn)
+			got, gotErr := headersPolicyService(test.dhp, test.in, false, test.dyn)
+			assert.Equal(t, test.want, got)
+			assert.Equal(t, test.wantErr, gotErr)
+		})
+	}
+}
+
+func TestAllowHostHeaderRewrite(t *testing.T) {
```

Comment:
Since we're using row-based testing here, it looks like we should be able to fold these test cases into `TestHeadersPolicy` -- `in` -> `hp`, can drop `dyn` entirely since it's not used in these test cases.